src/modules/ekf2/ekf2_main.cpp

```diff
@@ -2496,7 +2556,7 @@ int Ekf2::task_spawn(int argc, char *argv[])
 	_task_id = px4_task_spawn_cmd("ekf2",
 				      SCHED_DEFAULT,
 				      SCHED_PRIORITY_ESTIMATOR,
-				      6600,
+				      7100,
```

Comment:
@TSC21 I'm assuming you have to raise this number until the stack is big enough?

Reply:
Right but is that the right procedure in these cases?

Reply:
I think so, what else could you do?

Reply:
At this stage, I think nothing. I already simplified the code the max I could. Unless I am missing another simplification which is not obvious. I will increase the stack to `7600` just in case and ask @testflights to have a go again.

Reply:
@TSC21 should we give it another go?

Reply:
@Tony3dr I need to rebase first and increase the stack size a bit.

Reply:
@TSC21 can you please investigate why a stack increase is required? Seems like a regression to me.

Reply:
@bkueng at this stage I am not aware of any other solution, though I agree with you.

Reply:
The new ecl/EKF helpers you're using require quite a bit of stack because the entire covariance matrix is copied to create a temporary matrix::SquareMatrix. To avoid the copy we could work towards using a SquareMatrix directly in ecl/EKF (instead of a 2d array) or perhaps see if the matrix library could be made to optionally work with references rather than an internal copy. 

Anyway, I'd still see how much additional memory we'd actually need here. It might be fine for now.

Reply:
> the matrix library could be made to optionally work with references rather than an internal copy.

I think we could explore that but atm I think we can get this in with a bit of a stack increase and then iterate over it. Are you ok with that @bkueng?

Reply:
Not so close before a release.
Also I see a stack usage of 4568 bytes on master (with position initialized), whereas it is 6744 with this PR, so more than 2 KB increase. Can you please look into alternatives?

Reply:
I guess I can but I really wanted to bring this to the release but I don't know how much time more I can spend around this so to actually make it through. This implementation already took a lot of time to make possible, so I guess I will leave it as is and get back to it later.

Reply:
@TSC21 one quick thing you could try is to only grab a single copy of the covariances (_ekf.coveriances()) and compute the rest in place (instead of _ekf.covariances_diagonal(), _ekf.orientation_covariances(), _ekf.velocity_covariances(), _ekf.position_covariances()).

The core problem is each of these calls is creating a new temporary SquareMatrix. Instead, do the work once then on that same SquareMatrix grab the diagonal and do the slice operations as needed.

Reply:
Done!